clusterpools/apply.sh

```diff
@@ -510,14 +594,88 @@ if [[ "$CLUSTERPOOL_NAME" == "" ]]; then
     if [[ "$INPUT_CLUSTERPOOL_NAME" == "" ]]; then
         CLUSTERPOOL_NAME="$GENERATED_CLUSTERPOOL_NAME-$(echo $PLATFORM | tr '[:upper:]' '[:lower:]')-$(echo $CLUSTERIMAGESET_NAME | sed 's/\.//g')"
     else
+        CUSTOM_CLUSTERPOOL_NAME="true"
         CLUSTERPOOL_NAME="$INPUT_CLUSTERPOOL_NAME"
     fi
 fi
 printf "${GREEN}* Using Name: $CLUSTERPOOL_NAME${CLEAR}\n"
 
 
+#-----OPTIONALLY CONFIGURE A CUSTOM INSTALL CONFIG SECRET-----#
+if [[ "$YQ_INSTALLED" != "false" ]]; then
+    if [[ "$CUSTOM_CLUSTERPOOL_NAME" == "true" ]]; then
+        CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME="${CLUSTERPOOL_NAME}-${RANDOM_IDENTIFIER}-install-config-template"
+    else
+        CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME="${CLUSTERPOOL_NAME}-install-config-template"
+    fi
+    if [[ "$CLUSTERPOOL_INSTALL_CONFIG_FILE" == "" ]]; then
+        printf "${BLUE}- note: to skip this step in the future and create a new install-config secret from a file automatially, export CLUSTERPOOL_INSTALL_CONFIG_FILE${CLEAR}\n"
```

Comment:
Maybe this is just me, but calling this a "secret" is confusing to me since in my mind configurations are generally not secrets, and I started wondering if it needed my OCP pull secret.

I propose leaving off the word "secret" in this section except perhaps to make it clear that the resource is being stored as a secret.

Reply:
@dhaiducek Ahh yes, so I named the input var with the `_SECRET_NAME` at the end because we specifically want the user to provide us the _name_ of install config secret on the cluster that they want us to use to override this.  This variable otherwise holds the generated name of the secret we'll create on the cluster and point hive to.  Its confusing to me primarily because hive uses secrets (as opposed to configmaps) to hold this info!

I would propose we keep the secret naming to keep the resource type obvious!

Reply:
I agree about the variable name. I'm talking about the actual CLI output:
Maybe remove some of the "secret" references in here. I think mainly this one (since we're already listing secrets and we've referred to them as secrets previously):
```
Create a new Install Config (interactive).
Enter the number corresponding to your desired Install Config from the list above
```
or:
```
Create a new Install Config (interactive).
Enter the number corresponding to your desired secret containing the Install Config from the list above
```

Reply:
Ahhh, good point there - let me clear that up!

Reply:
Oh oh oh - @dhaiducek this line:
```
printf "${BLUE}- note: to skip this step in the future and create a new install-config secret from a file automatially, export CLUSTERPOOL_INSTALL_CONFIG_FILE${CLEAR}\n"
```
is telling the user that if they set CLUSTERPOOL_INSTALL_CONFIG_FILE, we'll read in the contents of that file and create and use a secret without their input!

Reply:
Right--that's the line that I'm thinking it's okay to leave alone. It's the others inside of the creation portion.

Reply:
But it was an initial confusion I had that I no longer have, and others may not have the same misunderstanding I did.